### PR TITLE
Refactored fire_act

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Refactored fire_act() to be in line with TG version, removed useless parameter, added signal, made non sleepable and forced to call parent.
Added atom_act.dm file for the various *_act procs.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

**Review by DreamySkrell — APPROVED** [review]

File: code/game/atom/atom_act.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * If you're adding something here, you likely want a signal and SHOULD_CALL_PARENT(TRUE)
+ * +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+ */
+
+/**
+ * Respond to fire being used on our atom
+ *
+ * Default behaviour is to send [COMSIG_ATOM_FIRE_ACT] and return
+ */
+/atom/proc/fire_act(exposed_temperature, exposed_volume)
```
Comment (DreamySkrell) [inline code/game/atom/atom_act.dm:14]: would be nice to comment what `exposed_temperature` and `exposed_volume` are supposed to mean
is temperature in kelvins? volume in...?

Comment (Arrow768) [reply code/game/atom/atom_act.dm:14]: The variables should be documented.
If someone who is unfamiliar with the system looks at exposed_temperature and exposed_volume they should be able to look at a description to find out what purpose those parameters have and what values they can expect to receive.

It not being documented on TG is not relevant for us.
If it bothers you, you can always PR the description to tg as well.

**Review by Arrow768 — CHANGES_REQUESTED** [review]
Other than documentation thats fine

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**